update(Row): Add inline prop.

stefhatcher opened the pull request:
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

Adds an inline prop, switching the row from being display flex, to flex-inline. When it's inline, its column spacing is the same as compact (8px).

## Motivation and Context

Need it! Handy for after content to be right up against main content with middle alignment as an option!

## Testing

- added test
- added story

## Screenshots

<img width="865" alt="Screen Shot 2019-07-23 at 4 55 11 PM" src="https://user-images.githubusercontent.com/306275/61755127-0e07aa00-ad6b-11e9-9e7a-5ab34c25adbd.png">

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

milesj commented:
`flex-inline` is super buggy in some browsers. That's pretty much why we haven't used it in other places.

stefhatcher commented:
i've been anit-`inline-flex` in the past, but it's support looks good now and can't be as funky as switching everything to inline-block and v-align just for inlining. inline block introduces a bunch of funk. i could switch it to the even newer `inline-grid` :D

stefhatcher commented:
it has pretty great global usage now:
<img width="1248" alt="Screen Shot 2019-07-24 at 9 43 01 AM" src="https://user-images.githubusercontent.com/306275/61811977-7a2fef80-adf7-11e9-8abc-1f6257ad9da3.png">

milesj commented:
It's not so much usage, but broken implementations. For example, all buttons and inline-block elements were `inline-flex` when we originally started this UI, and they were so problematic we had to revert them.

stefhatcher commented:
checked out what dls does and they use inline-flex as well, but they put it on the children and keep the parent flex. updated to match that! works well. also cleaned up the inner spacing to be consistent with Grid and AdaptiveGrid gap/gutter.

stefhatcher commented:
bump @milesj

milesj commented on packages/core/src/components/Row/index.tsx:
```diff
@@ -65,13 +69,18 @@ export class Row extends React.Component<Props & WithStylesProps> {
-        <div className={cx(styles.primary, truncated && styles.primary_truncated)}>{children}</div>
-
-        {after && <div className={cx(compact ? styles.after_compact : styles.after)}>{after}</div>}
+        {before && <div className={cx(styles.before, inline && styles.inline)}>{before}</div>}
```
I don't believe we should remove the compact padding in this PR. Also seems weird to only have 8px padding on spacious (28px) rows.

stefhatcher replied:
the 8px is to get the column spacing to be consistent with other column spacing in Grid and AdaptiveGrid. i can do it in a separate PR or keep it here

milesj replied:
Should go through design approval first.

stefhatcher replied:
talked with design - they are 👍 with this change. will add the larger column style for `spacious` too

stefhatcher commented:
updates made based on design feedback @milesj
